### PR TITLE
feat(input) : Add tests for Input Select

### PR DESCRIPTION
**Pull request by udhadpa**
Hi All,
Please review the tests written for select input component. It covers the basic CSS styling as per the redlines given in the design doc.
I have added two tests 
1. Basic Select Input Test
2.Basic Select Input Test in Focus state
both for desktop and mobile.
@StommePoes @JasonLantz @wyseguyonline @mhomolak @umahaea 
Thanks,

### CODE REVIEW

File: src/test/java/elementsTests/InputsTest.java [hunk 1]
```diff
@@ -775,6 +782,135 @@ private void basicInputErrorFocusTest(String[] expBoxShadow, String[] expBorder,
...
+    public Object[][] getBasicSelectInputFocusData() {
+        return new Object[][]{
+                {"1px", "solid", new String[]{commonUtils.hex2Rgb("#047A9C"), commonUtils.hex2RgbWithoutTransparency("#047A9C")}, "rgb(4, 122, 156) 0px 0px 5px 0px", "14px", commonUtils.hex2Rgb("#252525"), "18px"}
+        };
+    }
+
+    @Test(testName = "Basic Select Input Test Focus", dataProvider = "Basic Select Input Test Data Focus", groups = {"desktop-regression"})
+    private void basicSelectInputFocusTest(String expBorderWidth, String expBorderStyle, String[] expBorderColor, String expBoxShadow, String expOptionFontSize, String expOptionColor, String expOptionLineHt) throws InterruptedException {
+        if (browser.equals("firefox") || browser.equals("safari") || browser.equals("ie") || browser.equals("edge") || lBrowser.equals("firefox")) {
+            throw new SkipException("the focus operation is not supported on firefox/safari/ie drivers");
```
Comment (StommePoes) [inline src/test/java/elementsTests/InputsTest.java]: technically edge is a completely different browser than IE even though both are Microsoft :)

File: src/test/java/elementsTests/InputsTest.java [hunk 2]
```diff
@@ -775,6 +782,135 @@ private void basicInputErrorFocusTest(String[] expBoxShadow, String[] expBorder,
...
+        boxShadow = commonUtils.getCSSValue(inputsPgObj.basicSelectInput, "box-shadow");
+        isBoxShadow = commonUtils.assertValue(boxShadow, expBoxShadow, "Box-shadow of Select Input (Focus) is not as per spec");
+        for (By elem : options) {
+            Select selectByValue = new Select(driver.findElement(By.id("w")));
+            selectByValue.selectByIndex(i);
+            basicInputFontSize = commonUtils.getCSSValue(elem, "font-size");
+            basicInputValueColor = commonUtils.getCSSValue(elem, "color");
+            basicInputLineheight = commonUtils.getCSSValue(elem, "line-height");
+
+            isBasicInputFontSize = commonUtils.assertValue(basicInputFontSize, expOptionFontSize, "Font-size of Option " + i + " is not as per spec ");
```
Comment (StommePoes) [inline src/test/java/elementsTests/InputsTest.java]: we have no CSS for the options, nor do I believe we can style them. In addition, in the actual CSS code of the selects, I see references to pe-select-bg and stuff but I can't find where that's defined. @mhomolak can you explain?

File: src/test/java/elementsTests/InputsTest.java [hunk 3]
```diff
@@ -1279,6 +1415,125 @@ private void basicInputErrorFocusMobileTest(String[] expBoxShadow, String[] expB
...
+        actIconClass = commonUtils.getAttributeValue(inputsPgObj.basicSelectInputIcon, "class", "mobile");
+        isIconClass = commonUtils.assertValue(actIconClass, "pe-icon--dropdown-open-18", "Dropdown icon does not comply to the \"pe-icon--dropdown-open-18\" spec at mode " + mode);
+
+        Assert.assertTrue(isBasicInputHeight && isPaddingLeft && isPaddingRight && isMarginTop && isBasicInputFontSize && isBasicInputLineheight && isBasicInputValueColor && isBackgroundColor && isLabelColor && isLabelFontSize && islabelLineHeight && isLabelFor && isIconClass);
+    }
+
+    @Test(testName = "Mobile : Basic Select Input Test Focus", dataProvider = "Basic Select Input Test Data Focus", groups = {"mobile-regression"})
+    private void basicSelectInputFocusMobileTest(String expBorderWidth, String expBorderStyle, String[] expBorderColor, String expBoxShadow, String expOptionFontSize, String expOptionColor, String expOptionLineHt) throws InterruptedException {
+        if (setAppium.equals("iOS")) {
+            throw new SkipException("the focus-box shadow operation is not supported on iOS");
```
Comment (StommePoes) [inline src/test/java/elementsTests/InputsTest.java]: Hey everyone, for this reason I continue to propose adding an outline color that matches what the box shadow would be, unless that's indistinguishable on iOS from the border-color change we do. We are currently suppressing the outline from the browser and I wonder if this is a smart thing to do?

File: src/test/java/elementsTests/InputsTest.java [hunk 4]
```diff
@@ -1279,6 +1415,125 @@ private void basicInputErrorFocusMobileTest(String[] expBoxShadow, String[] expB
...
+        boxShadow = commonUtils.getCSSValue(inputsPgObj.basicSelectInput, "box-shadow", "mobile");
+        isBoxShadow = commonUtils.assertValue(boxShadow, expBoxShadow, "Box-shadow of Select Input (Focus) is not as per spec ");
+        for (By elem : options) {
+            Select selectByValue = new Select(appium.findElement(By.id("w")));
+            selectByValue.selectByIndex(i);
+            basicInputFontSize = commonUtils.getCSSValue(elem, "font-size", "mobile");
+            basicInputValueColor = commonUtils.getCSSValue(elem, "color", "mobile");
+            basicInputLineheight = commonUtils.getCSSValue(elem, "line-height", "mobile");
+
+            isBasicInputFontSize = commonUtils.assertValue(basicInputFontSize, expOptionFontSize, "Font-size of Option " + i + " is not as per spec ");
```
Comment (StommePoes) [inline src/test/java/elementsTests/InputsTest.java]: Again I dunno that we have any control over these option styles. If things happen to pass now, that does not mean things will pass in the future. Given how Android uses some kind of rolodex thing and my Windows phone's Edge uses a dropdown set to my font and contrast needs, I dunno that it's useful to test these, as they'll differ per device.

**Comment by StommePoes** [comment]
select on IE11

I know the tests can't catch this, but I think everyone on the team should be aware that this is what we're stuck for IE with until we try something like the <a href="https://github.com/filamentgroup/select-css">filament group hack</a>. There's nothing wrong with Mike's code, it's just the nature of OSes having control over form controls. Similarly Payal had to skip the lack of box-shadow on iOS and she can't test focus states with Selenium on almost all browsers (I only see Chrome/chromium missing from her list).

File: src/test/java/elementsTests/InputsTest.java (hunk 2 above)
Comment (mhomolak) [inline src/test/java/elementsTests/InputsTest.java]: Everything in _select-box.scss was leftovers from v0.  None of that is referenced in the select input.

File: src/test/java/elementsTests/InputsTest.java (hunk 3 above)
Comment (mhomolak) [inline src/test/java/elementsTests/InputsTest.java]: I thought box-shadow was no longer an issue on ios?  I'm seeing support starting in v 9.3.  The only browser left that does not support it is Opera which isn't surprising given their lack of support for numerous things in css.

File: src/test/java/elementsTests/InputsTest.java (hunk 1 above)
Comment (umahaea) [inline src/test/java/elementsTests/InputsTest.java]: Jus want to mention that these are limitations on the tool for these browsers and this kind of operation. We have tested this manually to ensure the focus is working.

File: src/test/java/elementsTests/InputsTest.java (hunk 3 above)
Comment (udhadpa) [inline src/test/java/elementsTests/InputsTest.java]: Hi Mike, when we run our tests, the value for box shadow returned is "none" for all iOS devices.

**Comment by mhomolak** [comment]
Yes, Eajaz explained to me that the problem is with the tool being used for
testing, not the browser.

**Comment by mhomolak** [comment]
I'll work on implementing a fix for the IE issue.

**Comment by umahaea** [comment]
apart from Mallory's comments, looks good to me..

**Comment by umahaea** [comment]
Merging. Got enough approvals. As discussed, Mike will work on the IE fix on a seperate branch.

**Pull request merged**